packages/next/src/server/app-render/encryption.ts

```diff
@@ -120,7 +120,10 @@ export async function decryptActionBoundArgs(
   // This extra step ensures that the server references are recovered.
   const serverModuleMap = getServerModuleMap()
   const transformed = await decodeReply(
-    await encodeReply(deserialized),
+    await encodeReply(deserialized, {
+      // TODO: How is decryptActionBoundArgs used? Do we need to support temporary references here?
+      temporaryReferences: undefined,
+    }),
```

Comment:
The missing piece is how we handle temporary references in `decryptActionBoundArgs`. Hopefully @shuding could shed some light on this. Since we don't create a response immediately we either need to return the temporary references or accept them as a function arg. But this seems all hidden behind a compiler so I'm wondering if I can test plain JS first to settle on a target and then adjust the compiler to the new target.

Reply:
I don't think we need temporary references for this.

The extra serialize/deserialize isn't representing a transfer from server to client and what you get out of the round trip should be (ideally equivalent) to if you had just not done the serialization in the first place. There really isn't a "client" to be providing a reference from that we can refer to when decoding the the bound args.

I'm pretty sure you can just omit the temporaryReferences option here

Reply:
Agreed with @gnoff here where encrypted values are same as server data that is hard coded into the action during its creation (fully server side). It's not something gets sent from the client that will be returned back.